recipes/cimg/all/conanfile.py

```diff
@@ -1,64 +1,114 @@
-        # ("enable_xshm",         "cimg_use_xshm",       "xshm/???"),
-    )
-    # TODO: Update requirements when available in CCI
 
-    options = dict((option, [True, False]) for option, _, _ in _cimg_options)
-    default_options = dict((option, True) for option, _, _ in _cimg_options)
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "enable_fftw": [True, False],
+        "enable_jpeg": [True, False],
+        "enable_openexr": [True, False],
+        "enable_png": [True, False],
+        "enable_tiff": [True, False],
+        "enable_ffmpeg": [True, False],
+        "enable_opencv": [True, False],
+        "enable_magick": [True, False],
+        "enable_xrandr": [True, False],
+        "enable_xshm": [True, False],
+    }
+    default_options = {
+        "enable_fftw": True,
+        "enable_jpeg": True,
+        "enable_openexr": True,
+        "enable_png": True,
+        "enable_tiff": True,
+        "enable_ffmpeg": True,
+        "enable_opencv": True,
+        "enable_magick": False,
+        "enable_xrandr": False,
+        "enable_xshm": False,
+    }
 
-    def configure(self):
-        tools.check_min_cppstd(self, "11")
+    no_copy_source = True
 
     @property
     def _source_subfolder(self):
         return "source_subfolder"
 
+    @property
+    def _cimg_defines(self):
+        return [
+            ("enable_fftw",    "cimg_use_fftw"),
+            ("enable_jpeg",    "cimg_use_jpeg"),
+            ("enable_openexr", "cimg_use_openexr"),
+            ("enable_png",     "cimg_use_png"),
+            ("enable_tiff",    "cimg_use_tiff"),
+            ("enable_ffmpeg",  "cimg_use_ffmpeg"),
+            ("enable_opencv",  "cimg_use_opencv"),
+            ("enable_magick",  "cimg_use_magick"),
+            ("enable_xrandr",  "cimg_use_xrandr"),
+            ("enable_xshm",    "cimg_use_xshm"),
+        ]
+
     def requirements(self):
-        for option, _, req in self._cimg_options:
-            if getattr(self.options, option):
-                self.requires(req)
+        if self.options.enable_fftw:
+            self.requires("fftw/3.3.9")
+        if self.options.enable_jpeg:
+            self.requires("libjpeg/9d")
+        if self.options.enable_openexr:
+            self.requires("openexr/2.5.7")
+        if self.options.enable_png:
+            self.requires("libpng/1.6.37")
+        if self.options.enable_tiff:
+            self.requires("libtiff/4.3.0")
+        if self.options.enable_ffmpeg:
+            self.requires("ffmpeg/4.4")
+        if self.options.enable_opencv:
+            self.requires("opencv/4.5.3")
+        if self.options.enable_magick:
+            self.requires("imagemagick/7.0.11-14")
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, "11")
+        # TODO: Update requirements when available in CCI
+        if self.options.enable_xrandr:
```

Comment:
these two are part of xorg IIRC

Reply:
xrandr I know it's part of xorg due the numerous time I had problems on my Linux.